### ISSUE TITLE
Upload size is limited by wolkenkit-proxy

### ISSUE THREAD

**Issue by mattwagl** [thenativeweb/wolkenkit#151]
### What is this bug about?

`wolkenkit-depot-file` allows you to store large files. However since the introduction of `wolkenkit-proxy` the upload size seems to be limited to 1MB.

### What steps are needed to reproduce the bug?

- Start a wolkenkit application
- Upload a file larger than 1MB using the `wolkenkit-depot-client-js` from Node
- Catch the error and log the output

### What is the expected result?

- `addFile` should not throw an error. 
- The file should be uploaded.

### What is the actual result?

- The wolkenkit-application responds with status 413 and the following result:

````html
<html>\r\n<head><title>413 Request Entity Too Large</title></head>\r\n<body bgcolor="white">\r\n<center><h1>413 Request Entity Too Large</h1></center>\r\n<hr><center>nginx/1.14.2</center>\r\n</body>\r\n</html>\r\n
````

### What else should we know?

- I've create a repo called [depot-test](https://github.com/mattwagl/depot-test) to make this reproducible
- This problem seems to be caused by the nginx proxy where the `client_max_body_size` defaults to 1MB. This option should be added only to the nginx server that is responsible for the depot. When set to `0` the check is omitted.
- Additionally `wolkenkit-depot-client-js` needs to be adjusted since `axios` also has an option called `maxContentLength`. This option seemed to cause errors when uploading files larger than 10mb. If you set the option to `Infinity` large uploads e.g. with 500mb are possible.
- This bug was reported by @SteffenGottschalk, and @timrach. So we should notify them once this has been done.

**Comment by mattwagl** [comment]
That's the fix I applied to `nginx.conf.tpl` in `wolkenkit-proxy`:

```nginx
...
server {
  server_name ${DEPOT_EXTERNAL_HOST};
  listen ${DEPOT_EXTERNAL_PORT} ssl http2;

  ssl on;
  ssl_certificate ${DEPOT_CERTIFICATE};
  ssl_certificate_key ${DEPOT_PRIVATE_KEY};
  client_max_body_size 0;
...
```

**Comment by mattwagl** [comment]
And this is the fix I applied to `DepotClient.js` inside of `wolkenkit-depo-client-js`:

```javascript
...
    try {
      await request({
        method: 'post',
        url: `${protocol}://${host}:${port}/api/v1/add-file`,
        data: content,
        maxContentLength: Infinity,
        headers,
        onUploadProgress: progressEvent => {
          this.processProgress({ id, type: 'upload', startTime, progressEvent });
        }
      });
...
```

**Comment by quarterpi** [comment]
Is this issue still open? If so, what’s left to do?

**Comment by goloroden** [comment]
@quarterpi AFAICS yes the bug is still open.

@mattwagl Could you maybe elaborate on what the state of this is? I only remember that you looked at it, but then aborted working on it. What's the current status here?

**Comment by quarterpi** [comment]
Probably better to look for a different 1st issue then 😉. Thank you for the prompt reply.

**Comment by goloroden** [comment]
You’re welcome 😊 

Since we are making dramatic changes to the codebase for 4.0, it’s currently a bit difficult to tell a good first issue from this list. 

But if you want to, it would be great to welcome you in the wolkenkit Slack at http://slackin.wolkenkit.io/ and then we could discuss ideas there. What do you think?

**Comment by quarterpi** [comment]
I’m already there... I’ll contact you.

**Comment by mattwagl** [comment]
@goloroden @quarterpi To report the current status: I reported the bug but I didn't start working on it. I've created the test repository as a way to track down the cause of the issue for possible contributors. The changes I made to `wolkenkit-proxy` where just temporary in order to check if that is the right place to fix the issue. So in theory a good first issue but not right now since we might switch from nginx to haproxy.
